editor/export/codesign.cpp

```diff
@@ -102,61 +102,63 @@ void CodeSignCodeResources::add_rule2(const String &p_rule, const String &p_key,
 }
 
 CodeSignCodeResources::CRMatch CodeSignCodeResources::match_rules1(const String &p_path) const {
-	CRMatch found = CRMatch::CR_MATCH_NO;
-	int weight = 0;
-	for (int i = 0; i < rules1.size(); i++) {
-		RegEx regex = RegEx(rules1[i].file_pattern);
-		if (regex.search(p_path).is_valid()) {
-			if (rules1[i].key == "omit") {
-				return CRMatch::CR_MATCH_NO;
-			} else if (rules1[i].key == "nested") {
-				if (weight <= rules1[i].weight) {
-					found = CRMatch::CR_MATCH_NESTED;
-					weight = rules1[i].weight;
-				}
-			} else if (rules1[i].key == "optional") {
-				if (weight <= rules1[i].weight) {
-					found = CRMatch::CR_MATCH_OPTIONAL;
-					weight = rules1[i].weight;
-				}
-			} else {
-				if (weight <= rules1[i].weight) {
-					found = CRMatch::CR_MATCH_YES;
-					weight = rules1[i].weight;
-				}
-			}
-		}
-	}
-	return found;
+    CRMatch found = CRMatch::CR_MATCH_NO;
+    int weight = 0;
+    
+    for (const auto& rule : rules1) {
```

Comment:
We do not allow `auto`, see https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword